core/src/test/resources/QualificationExpectations/complex_dec_expectation.csv

```diff
@@ -1,2 +1,2 @@
 App Name,App ID,Recommendation,Estimated GPU Speedup,Estimated GPU Duration,Estimated GPU Time Saved,SQL DF Duration,SQL Dataframe Task Duration,App Duration,GPU Opportunity,Executor CPU Time Percent,SQL Ids with Failures,Unsupported Read File Formats and Types,Unsupported Write Data Format,Complex Types,Nested Complex Types,Potential Problems,Longest SQL Duration,NONSQL Task Duration Plus Overhead,Unsupported Task Duration,Supported SQL DF Task Duration,Task Speedup Factor,App Duration Estimated,Unsupported Execs,Unsupported Expressions,Estimated Job Frequency (monthly)
-"Spark shell","local-1626104300434","Not Recommended",1.0,129898.52,1205.47,2429,1469,131104,1923,88.35,"","","","struct<firstname:string,middlename:array<string>,lastname:string>;struct<current:struct<state:string,city:string>,previous:struct<state:map<string,string>,city:string>>;array<struct<city:string,state:string>>;map<string,string>;map<string,array<string>>;map<string,map<string,string>>;array<array<string>>;array<string>","struct<firstname:string,middlename:array<string>,lastname:string>;struct<current:struct<state:string,city:string>,previous:struct<state:map<string,string>,city:string>>;array<struct<city:string,state:string>>;map<string,array<string>>;map<string,map<string,string>>;array<array<string>>","NESTED COMPLEX TYPE",1260,128847,306,1163,2.68,false,"CollectLimit","",30
+"Spark shell","local-1626104300434","Not Recommended",1.0,130359.57,744.42,1500,1469,131104,1187,88.35,"","","","struct<firstname:string,middlename:array<string>,lastname:string>;struct<current:struct<state:string,city:string>,previous:struct<state:map<string,string>,city:string>>;array<struct<city:string,state:string>>;map<string,string>;map<string,array<string>>;map<string,map<string,string>>;array<array<string>>;array<string>","struct<firstname:string,middlename:array<string>,lastname:string>;struct<current:struct<state:string,city:string>,previous:struct<state:map<string,string>,city:string>>;array<struct<city:string,state:string>>;map<string,array<string>>;map<string,map<string,string>>;array<array<string>>","NESTED COMPLEX TYPE",1260,129598,306,1163,2.68,false,"CollectLimit","",30
```

Comment:
I think gpu opportunity changed  a lot here, can we validate this makes sense?

Reply:
This eventlog has less computation time and high overhead. 
```
jobOverheadTime = 129314
appDuration: 131104
appDurationNoOverhead = 1790
```
Earlier estimation would multiply GPU ratio (based on supportedtaskdurations and SQLDFTaskDurations) with sumofSQLDurations (wallclock). In this case the sum was higher. 
`sparkSQLDFWallClockDuration: 2429 (130, 479, 65, 65, 259, 76, 95, 1260)`

In this PR, we multiply GPU ratio(based on taskDurations of taskTimes) with appDurationNoOverhead.
`sqlDataFrameDurationToUse = 1500 (1790*0.83)`

This is the reason for the change in estimated GPU duration and it makes sense as earlier we weren't considering the overall taskDurations in the ratio calculations and also overestimating the sparkSQLDFWallClock durations.

Reply:
ok great